index.js

```diff
@@ -33,32 +33,19 @@ module.exports = function (opts = {}) {
 
     app.authboot = {};
     const lookup = app.authboot.lookup = (lookupOpt || function ({ name, password }, callback) {
-      const pass = users.get(name);
-      if (!pass || !compare(pass, password)) {
-        debug('Invalid password for valid username %s', name);
-        const error = new Error('Invalid password for given username');
-        error.status = 401;
-        return callback(error);
-      }
-      return callback(null, true);
+      const hash = users.get(name);
+      if (!hash) return callback(null, false);
```

Comment:
why false and not return an error?

Reply:
@msluther this is so the `express-basic-auth` middleware can correctly send the configured `unauthorizedResponse` rather than it assume it will be handled by some error middleware that you might not have in your express app.

Reply:
Ah. Looks like `bcrypt.compare` does the same thing.  Hadn't realized that. https://github.com/kelektiv/node.bcrypt.js#to-check-a-password